src/Microsoft.VisualStudio.Composition/Reflection/ResolverExtensions.cs

```diff
@@ -162,19 +162,15 @@ internal static void GetInputAssemblies(this TypeRef typeRef, ISet<AssemblyName>
                     GetInputAssemblies(typeArg, assemblies);
                 }
 
+                assemblies.Add(typeRef.AssemblyName);
+
                 // Base types may define [InheritedExport] attributes or otherwise influence MEF
                 // so we should include them as input assemblies.
                 // Resolving a TypeRef is a necessary cost in order to identify the transitive closure of base types.
-                var type = typeRef.Resolve();
-                foreach (var baseType in type.EnumTypeAndBaseTypes())
-                {
-                    assemblies.Add(baseType.GetTypeInfo().Assembly.GetName());
-                }
-
-                // Interfaces may also define [InheritedExport] attributes, metadata view filters, etc.
-                foreach (var iface in type.GetTypeInfo().GetInterfaces())
+                //var type = typeRef.Resolve();
+                foreach (var baseType in typeRef.BaseTypes)
                 {
-                    assemblies.Add(iface.GetTypeInfo().Assembly.GetName());
+                    assemblies.Add(baseType.ResolvedType.GetTypeInfo().Assembly.GetName());
```

Comment:
>assemblies.Add(baseType.ResolvedType.GetTypeInfo().Assembly.GetName()) [](start = 20, length = 70)

Why is resolving the TypeRef necessary, consider TypeRef has a property that stores the assembly name?